dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -570,14 +576,42 @@ private void checkToolTokens(User authUser, Long userId, String organization) {
         return userDAO.findAll();
     }
 
+    /**
+     * TODO: Use enum for the source parameter
+     * @param user      The Authorized user
+     * @param source    token source, currently either the google or github TokenType
+     * @return          The updated user
+     */
     @GET
     @Timed
     @UnitOfWork
     @Path("/user/updateUserMetadata")
     @ApiOperation(value = "Update metadata for logged in user", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, notes = "Update metadata for logged in user.", response = User.class)
-    public User updateLoggedInUserMetadata(@ApiParam(hidden = true) @Auth User user) {
+    public User updateLoggedInUserMetadata(@ApiParam(hidden = true) @Auth User user, @ApiParam(value = "Token source", allowableValues = "google.com, github.com") @QueryParam("source") TokenType source) {
         User dbuser = userDAO.findById(user.getId());
-        dbuser.updateUserMetadata(tokenDAO);
+        if (source.equals(TokenType.GOOGLE_COM) || source.equals(null)) {
```

Comment:
confirm that a null source should be assumed to be google?

Reply:
null source means it'll try to update both